activerecord/lib/active_record/result.rb

```diff
@@ -141,9 +141,11 @@ def freeze # :nodoc:
     end
 
     private
-      def column_type(name, type_overrides = {})
+      def column_type(name, index, type_overrides)
         type_overrides.fetch(name) do
-          column_types.fetch(name, Type.default_value)
+          column_types.fetch(name) do
```

Comment:
`column_types` keys are indexes, so I think this line is not needed.

Reply:
Rechecked to ensure that isn't used with column name keys in different adapters. Removed. Thank you for the sugestion.